neo/rawio/axonarawio.py

```diff
@@ -298,9 +300,13 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
         if i_start is None:
             i_start = 0
         if i_stop is None:
-            i_stop = bin_dict['num_total_samples']
+            i_stop = self.get_signal_size(block_index=block_index, seg_index=seg_index)
         if channel_indexes is None:
-            channel_indexes = [i for i in range(bin_dict['num_channels'])]
+            channel_indexes = [i for i in range(self.header['signal_channels'].shape[0])]
+
+        if self.bin_file is None:
```

Comment:
I think having a constructed fake signal provided by default if no bin file was found is pretty dangerous, as the user might not be aware of this 'magic' happening in the background. This sort of interpolation should be rather an explicit preprocessing step in the analysis pipeline of the user.